python/lsst/ap/verify/ingestion.py

```diff
@@ -0,0 +1,480 @@
+
+    Returns
+    -------
+    metadata : `lsst.daf.base.PropertySet`
+        The full metadata from any Tasks called by this function.
+    """
+    # TODO: generalize to support arbitrary URIs (DM-11482)
+    log = lsst.log.Log.getLogger('ap.verify.ingestion.ingestDataset')
+    dataset.makeCompatibleRepo(repository)
+    log.info('Input repo at %s created.', repository)
+
+    metadata = dafBase.PropertySet()
+    metadata.combine(_ingestRaws(dataset, repository))
+    metadata.combine(_ingestCalibs(dataset, repository))
+    metadata.combine(_ingestTemplates(dataset, repository))
+    log.info('Data ingested')
+    return metadata
+
+
+def _ingestRaws(dataset, workingRepo):
+    """Ingest the raw data for use by LSST.
+
+    The original data directory shall not be modified.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset on which the pipeline will be run.
+    workingRepo : `str`
+        The repository in which temporary products will be created. Must be
+        compatible with ``dataset``.
+
+    Returns
+    -------
+    metadata : `lsst.daf.base.PropertySet`
+        The full metadata from any Tasks called by this function, or `None`.
+    """
+    rawRepo = _getOutputRepo(workingRepo, _INGESTED_DIR)
+    dataFiles = _getDataFiles(dataset.rawLocation)
+    return _doIngest(rawRepo, dataset.refcatsLocation, dataFiles)
+
+
+def _ingestCalibs(dataset, workingRepo):
+    """Ingest the raw calibrations for use by LSST.
+
+    The original calibration directory shall not be modified.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset on which the pipeline will be run.
+    workingRepo : `str`
+        The repository in which temporary products will be created. Must be
+        compatible with ``dataset``.
+
+    Returns
+    -------
+    metadata : `lsst.daf.base.PropertySet`
+        The full metadata from any Tasks called by this function, or `None`.
+    """
+    repo = _getOutputRepo(workingRepo, _INGESTED_DIR)
+    calibRepo = _getOutputRepo(workingRepo, _CALIBINGESTED_DIR)
+    calibDataFiles = _getCalibDataFiles(dataset.calibLocation)
+    defectFiles = _getDefectFiles(dataset.defectLocation)
+    return _doIngestCalibs(repo, calibRepo, calibDataFiles, defectFiles)
+
+
+def _ingestTemplates(dataset, workingRepo):
+    """Ingest the templates for use by LSST.
+
+    The original template repository shall not be modified.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset on which the pipeline will be run.
+    workingRepo : `str`
+        The repository in which temporary products will be created. Must be
+        compatible with ``dataset``.
+
+    Returns
+    -------
+    metadata : `lsst.daf.base.PropertySet`
+        The full metadata from any Tasks called by this function, or `None`.
+    """
+    # TODO: move doIngestTemplates to this module once DM-11865 resolved
```

Comment:
I like how you've handled template "ingestion" for now, this is a good placeholder for post- DM-11865.

Reply:
Per discussion with @jdswinbank, I think this is actually how we want `ap_verify` to handle templates in the long term. The AP pipeline will (probably) see the latest LSST data and the templates as a single repository, so we should try to simulate that.

Reply:
Interesting, OK!